pydatarecognition/main.py

```diff
@@ -63,109 +65,114 @@ def main(verbose=True):
         output_dir.mkdir()
     userdata = user_input_read(user_input)
     user_q, user_int = create_q_int_arrays(args, userdata)
+
     if args['jsonify']:
         for ciffile in ciffiles:
-            print(ciffile.name)
             json_data = cif_read_ext(ciffile, 'json')
             pre = Path(ciffile).stem
             json_dump(json_data, str(output_dir/pre) + ".json")
     else:
-        for ciffile in ciffiles:
-            if verbose:
-                ciflog.append(ciffile.name)
-            pcd = cif_read(ciffile)
-            try:
-                user_resampled, target_resampled = xy_resample(user_q, user_int, pcd.q,
-                                             pcd.intensity, x_step=args.get('qgrid_interval'))
-                corr_coeff = correlate(user_resampled[1], target_resampled[1])
-            except (AttributeError, ValueError) as e:
-                skipped_cifs.append((ciffile.name, e))
-                continue
+        if args['client'] == 'fs':

```

Comment:
This will work but it is almost certainly not the best way to do this.  What we are after is an interface where the code on the front side of the interface is the same regardless of the client.  It has functions like `find_one`, `find_many`, `update_one`, `update_many` and so on.  When `find_one` is called, the code checks what the client is, if it is `fs` it executes code that "finds one" by opening a file on the `fs`, but if it finds that the client is `mpc` it executes code that connects to mpc and fetches that thing.  This is also extensible, so in the future we could have a mongodb backend and this would work and so on.

If you want to see an example of how to structure this, please take a look at the code in regolith